app/screens/settings/display_theme/display_theme.test.tsx

```diff
@@ -157,4 +189,38 @@ describe('DisplayTheme', () => {
 
         expect(popTopScreen).toHaveBeenCalledTimes(1);
     });
+
+    it('should not allow user to select a theme rapidly', async () => {
+        const numOfSavePreferenceCalls = 1;
+        renderWithIntl(
+            <DisplayTheme
+                allowedThemeKeys={['denim', 'sapphire']}
+                {...displayThemeOtherProps}
+            />,
+        );
+
+        const sapphireTile = screen.getByTestId('theme_display_settings.sapphire.option');
+
+        fireEvent.press(sapphireTile);
+
+        await waitFor(() => {
+            expect(savePreference).toHaveBeenCalledWith(
+                expect.any(String),
+                expect.arrayContaining([
+                    expect.objectContaining({
+                        category: 'theme',
+                        value: expect.stringContaining('"type":"Sapphire"'),
+                    }),
+                ]),
+            );
+            expect(savePreference).toHaveBeenCalledTimes(numOfSavePreferenceCalls);
+        });
+
+        const denimTile = screen.getByTestId('theme_display_settings.denim.option');
+
+        fireEvent.press(denimTile);
+
+        // firing denimTile will not cause the savePreference to be called again since we have the prevent double tap
+        expect(savePreference).toHaveBeenCalledTimes(numOfSavePreferenceCalls);
```

Comment:
I wonder if I should make a similar test like this one, but add a delay of 1s (using fake timer), then click on the denimTile, it should show that the second tap is successful.